From 99871a421b585e63a2de4e033ac5fca8969cb0fe Mon Sep 17 00:00:00 2001
From: Peeyush Agarwal <apeeyush@google.com>
Date: Wed, 7 Mar 2018 16:21:50 +0000
Subject: [PATCH 152/306] ParceledListSlice shouldn't be created with null

Bug: 74242190
Change-Id: Ia539efb46bba0bab7a0bfdb0c9b4b2b09116acbf
Test: Reproduced the issue, then verified that this fixes it.
(cherry picked from commit db860eb1243c439c64e2c5a0205e06a4f1e49eb2)
---
 .../java/com/android/server/display/BrightnessTracker.java  | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/display/BrightnessTracker.java b/services/core/java/com/android/server/display/BrightnessTracker.java
index cabccf37eba..171f40ed37c 100644
--- a/services/core/java/com/android/server/display/BrightnessTracker.java
+++ b/services/core/java/com/android/server/display/BrightnessTracker.java
@@ -69,6 +69,7 @@ import java.nio.charset.StandardCharsets;
 import java.util.ArrayDeque;
 import java.util.ArrayList;
 
+import java.util.Collections;
 import java.util.Deque;
 import java.util.HashMap;
 import java.util.Map;
@@ -657,7 +658,10 @@ public class BrightnessTracker {
     }
 
     public ParceledListSlice<AmbientBrightnessDayStats> getAmbientBrightnessStats(int userId) {
-        return new ParceledListSlice<>(mAmbientBrightnessStatsTracker.getUserStats(userId));
+        ArrayList<AmbientBrightnessDayStats> stats = mAmbientBrightnessStatsTracker.getUserStats(
+                userId);
+        return (stats != null) ? new ParceledListSlice<>(stats) : new ParceledListSlice<>(
+                Collections.EMPTY_LIST);
     }
 
     // Not allowed to keep the SensorEvent so used to copy the data we care about.
-- 
2.17.1

